src/power_grid_model/core/serialization.py

```diff
@@ -0,0 +1,319 @@
+    """
+    result = JsonDeserializer(data).load()
+    assert_no_error()
+    return result
+
+
+def json_serialize(
+    data: Union[Mapping[str, np.ndarray], Mapping[str, Union[np.ndarray, Mapping[str, np.ndarray]]]],
+    dataset_type: Optional[str] = None,
+    use_compact_list: bool = False,
+    indent: int = 2,
+) -> str:
+    """
+    Dump data to a JSON str.
+
+    If the dataset_type is not specified or None, it will be deduced from the dataset if possible.
+    Deduction is not possible in case data is empty.
+
+    Args:
+        data: the dataset
+        dataset_type: the type of the dataset. Defaults to None. Required str-type if data is empty.
+        use_compact_list: whether or not to use compact lists (sparse data). Defaults to False.
+        indent:
+            use specified indentation to make data more readable
+            Use 0 or negative value for no indentation. Defaults to 2
+
+    Raises:
+        PowerGridError if there was an internal error.
+
+    Returns:
+        a serialized string containing the dataset
+    """
+    result = JsonSerializer(data=data, dataset_type=dataset_type).dump(use_compact_list=use_compact_list, indent=indent)
+    assert_no_error()
+    return result
+
+
+def msgpack_deserialize(data: bytes) -> Tuple[Mapping[str, Union[np.ndarray, Mapping[str, np.ndarray]]], str]:
+    """
+    Load serialized msgpack data to a new dataset.
+
+    Args:
+        data: the data to deserialize.
+
+    Raises:
+        ValueError if the data is inconsistent with the rest of the dataset or a component is unknown.
+        PowerGridError if there was an internal error.
+
+    Returns:
+        A tuple containing:
+            the deserialized dataset in Power grid model input format.
+            the type of the dataset.
+    """
+    result = MsgpackDeserializer(data).load()
+    assert_no_error()
+    return result
+
+
+def msgpack_serialize(
+    data: Union[Mapping[str, np.ndarray], Mapping[str, Union[np.ndarray, Mapping[str, np.ndarray]]]],
+    dataset_type: Optional[str] = None,
+    use_compact_list: bool = False,
+) -> bytes:
+    """
+    Dump the data to raw msgpack bytes.
+
+    If the dataset_type is not specified or None, it will be deduced from the dataset if possible.
+    Deduction is not possible in case data is empty.
+
+    Args:
+        data: the dataset
+        dataset_type: the type of the dataset. Defaults to None. Required str-type if data is empty.
+        use_compact_list: whether or not to use compact lists (sparse data). Defaults to False.
+
+    Raises:
+        KeyError if the dataset_type was not provided and could not be deduced from the dataset (i.e. it was empty).
+        PowerGridError if there was an internal error.
+
+    Returns:
+        a serialized string containing the dataset
+    """
+    result = MsgpackSerializer(data=data, dataset_type=dataset_type).dump(use_compact_list=use_compact_list)
+    assert_no_error()
+    return result
```

Comment:
please add helper functions to load/dump json/msgpack from/to files.

Reply:
what would you call them? we now have `(json|msgpack)_(deserialize|serialize)`

would you use `(json|msgpack)_(load|dump)`? for `msgpack` it's not really loading/dumping, but packing/unpacking, right?

Reply:
`(json|msgpack)_(deserialize|serialize)_(to|from)_file`